commands/root.go

```diff
@@ -129,45 +129,64 @@ const (
 __awless_get_all_ids()
 {
 		local all_ids_output
-		if all_ids_output=$(awless list infra --local --ids 2>/dev/null; awless list access --local --ids 2>/dev/null); then
+		local profile_args
+		if [[ "$1" != "" ]]; then
+			profile_args="-p $1"
+		else
+			profile_args="--local"
+	    fi
+		if all_ids_output=$(awless $profile_args list infra --ids 2>/dev/null; awless $profile_args list access --ids 2>/dev/null); then
 		COMPREPLY=( $( compgen -W "${all_ids_output[*]}" -- "$cur" ) )
 		fi
 }
 __awless_get_instances_ids()
 {
 		local all_ids_output
-		if all_ids_output=$(awless list instances --local --ids 2>/dev/null); then
+		local profile_args
+		if [[ "$1" != "" ]]; then
+			profile_args="-p $1"
+		else
+			profile_args="--local"
+	    fi
+		if all_ids_output=$(awless $profile_args list instances --ids 2>/dev/null); then
 		COMPREPLY=( $( compgen -W "${all_ids_output[*]}" -- "$cur" ) )
 		fi
 }
 __awless_get_conf_keys()
 {
-		local all_keys_output
-		if all_keys_output=$(awless config list --keys 2>/dev/null); then
+		local all_ids_output
```

Comment:
Here, for the completion of the command `awless config list ...`, the `-p` flag or `local` flag will have no effect. So there is no need to add the code here. The former implementation was enough.